pkg/sbom/node.go

```diff
@@ -211,14 +213,76 @@ func (n *Node) Equal(n2 *Node) bool {
 func (n *Node) flatString() string {
 	pairs := []string{}
 	n.ProtoReflect().Range(func(fd protoreflect.FieldDescriptor, v protoreflect.Value) bool {
-		pairs = append(pairs, string(fd.FullName())+":"+v.String())
+		switch fd.FullName() {
+		case "bomsquad.protobom.Node.external_references":
+			for _, ex := range n.ExternalReferences {
+				pairs = append(pairs, fmt.Sprintf("extref:%s", ex.flatString()))
+			}
+		case "bomsquad.protobom.Node.suppliers":
+			for _, i := range n.Suppliers {
+				pairs = append(pairs, fmt.Sprintf("supplier:%s", i.flatString()))
+			}
+		case "bomsquad.protobom.Node.originators":
+			for _, i := range n.Originators {
+				pairs = append(pairs, fmt.Sprintf("originator:%s", i.flatString()))
+			}
+		case "bomsquad.protobom.Node.identifiers":
+			for _, i := range n.Identifiers {
+				pairs = append(pairs, i.flatString())
```

Comment:
Is the GitHub linter grumpy because the `flatString()` function is recursive?

Reply:
I had forgotten to check in a file. Fixed now!